src/model/guild/member.rs

```diff
@@ -518,8 +518,7 @@ impl Member {
                 .guild(self.guild_id)?
                 .roles
                 .iter()
-                .filter(|(id, _)| self.roles.contains(id))
-                .map(|(_, v)| v.clone())
+                .filter_map(|(id, role)| self.roles.contains(id).then(|| role.clone()))
```

Comment:
I don't think this is more readable. Instead of two operations in sequence, there's now nesting which you have to pull apart mentally